### DISCUSSION TITLE
MatNWB calcium imaging without uploading raw images

### DISCUSSION THREAD

**Discussion by vikramvijayan** [dandi/helpdesk#63] [Q&A] [answered]
Hi,

I was wondering if there is a tutorial (similar to https://neurodatawithoutborders.github.io/matnwb/tutorials/html/ophys.html) but for uploading imaging data without uploading images/masks (just uploading different ROI df/f and raw signal). 

Any sample code would be useful! I could just upload as Timeseries data but I'm sure there is a better way to do this.

Thanks,
Vikram

**Comment by satra** [comment, marked as answer]
@vikramvijayan -  i'll let @bendichter answer the matnwb question, but at least from a DANDI perspective we would prefer the raw data for ophys. you can also include the different ROI df/f and raw signal.

**Reply by bendichter** [reply]
@satra yes, we would prefer the raw data, but this is actually a different question. In this case, Vikram wants to omit not only the raw data but also the ROI masks, i.e. the mappings that say which pixels of the image correspond to each of the ROIs. 

@vikramvijayan, there isn't a straightforward way to do this while retaining the `RoiResponseSeries` type. I could help you find a work-around, but can I ask why you don't want to store the image masks (or alternatively the pixel masks)?

**Reply by vikramvijayan** [reply]
@satra @bendichter : 

I would like to upload a large set of imaging data that I collected over the past several years. In the processing steps (done a while ago), I extracted raw signal from each ROI, and have been working with this raw signal since then. 

I still have all the raw images and masks, but the process of bringing these images back into the working data structure will be relatively intensive and error-prone. I'm changing my pipeline so that this is easier in the future, but I still want to be able to be able to upload these older datasets.

**Reply by vikramvijayan** [reply]
@bendichter let me know the possibilities when you get a chance (no rush). We could communicate through email if its easier!

**Reply by bendichter** [reply]
@vikramvijayan I discussed with the team and we think it would be better to include the ROI masks. There isn't a good way of storing an `RoiResponseSeries` without it, and the work-arounds are too hacky. I know this is a lot of work, but NWB says this is essential metadata and in this case I agree. If that is not possible, you can store this data is a `TimeSeries` data and indicate in the description that it is and roi response series. 

 Users would be able to access and use this data if they know that it is there, however the disadvantage of using TimeSeries is that your data will not be as findable on DANDI. DANDI pulls all the neurodata types into metadata, and users can query for all datasets that have an "RoiResponseSeres" type, which yours would not. You can mitigate this to some extent by mentioning RoiResponseSeries in the description of the dandiset.
